### DISCUSSION TITLE
Do we have a setting for fixed baseline images aside for only comparing from the previous run?

### DISCUSSION THREAD

**Discussion by faith-berroya** [FRSOURCE/cypress-plugin-visual-regression-diff#88] [Q&A] [answered]
Firstly, thank you for the great plugin! It works great on my local where I can always check for the image screenshots. However, I noticed that the screenshots were changing, and I have to constantly check if the plugin is getting the correct image or not because it always output as PASS even with different image, unless the image that will be compared has different dimension from the base image. 

I also have a question in here. Thank you so much!
https://github.com/FRSOURCE/cypress-plugin-visual-regression-diff/issues/87

**Comment by FRSgit** [comment, marked as answer]
Hey @faith-berroya,

I've responded in a linked issue - please, give me a sign whether that has helped you!

Regarding tests always passing - if you don't set `updateImages` to true every time the test is being run that shouldn't be the case. Plugin should catch any major changes in the page structure when it's visible on the screenshot. Can you elaborate more about your use-case and environment where you run the tests suite (OS, browser name, Cypress & Node versions)?

**Reply by faith-berroya** [reply]
Hi @FRSgit I set the `updateImages` to false in our main and is now not changing the images anymore. It works now. Thank you so much! 😄

**Reply by FRSgit** [reply]
Glad that it helped! 😄 Should I close the issue #87 as well?

**Reply by faith-berroya** [reply]
Yes please! Thank you so much! 😄

**Reply by notdotscott** [reply]
@FRSgit Apologies for jumping in here, but I think I have a similar request to @faith-berroya so didn't want to start a new thread (although happy to do so if you'd like to track it separately).

My issue is that I want to compare images from two dev environments, specifically to check whether changes in a sandbox have introduced any visual regressions compared to the staging environment. 

The way I've approached it is to generate a baseline image from stage, and then an image from the sandbox and compare the two. This all happens in the same test. Images aren't saved between test runs. 

I have it working with the following plugin: https://www.npmjs.com/package/@simonsmith/cypress-image-snapshot
(it's basically a fork of `cypress-image-snapshot` that works with Cypress 10).

I wanted to try and use your plugin instead as it's more active and up-to-date. However, because of the screenshot naming strategy you use internally, it will never compare the images generated using the above implementation.

Is there any way to get this to work without modifying your code?

**Reply by notdotscott** [reply]
As an example of what I'm attempting:

```javascript
describe('Compare', () => {
  it('target snapshot', () => {
    cy.visit(Cypress.env('TEST_TARGET_URL'));

    cy.matchImage();
  });

  it('source snapshot', () => {
    cy.visit(Cypress.env('TEST_SOURCE_URL'));

    cy.matchImage();
  });
});
```

**Reply by FRSgit** [reply]
Sorry for late response 😞 
@notdotscott That's and interesting usecase!
You're right - even if you pass the same `title` configuration option (as below) you'll get `#0` and `#1` suffix for each of the images.
```ts
cy.matchImage({ title: 'my-title' });
```

I'm not sure how the API for this feature should look like:
1. It could be done via additional option - I could let you pass `preventTitleClashes: false` and together with `title` option you could force tests to use the same title for screenshots. Resulting in comparing images. Your test would look like this:
```ts
const title = 'my-screenshot-title';
describe('Compare', () => {
  it('target snapshot', () => {
    cy.visit(Cypress.env('TEST_TARGET_URL'));

    cy.matchImage({ title, preventTitleClashes: false });
  });

  it('source snapshot', () => {
    cy.visit(Cypress.env('TEST_SOURCE_URL'));

    cy.matchImage({ title, preventTitleClashes: false });
  });
});
```
2. `.matchImage()` could return image details (like path) and that could be passed as `matchAgainst` option. Your test would look like this:
```ts
describe('Compare', () => {
  it('target snapshot', () => {
    cy.visit(Cypress.env('TEST_TARGET_URL'));

    cy.matchImage()
     .then(image => {
       cy.visit(Cypress.env('TEST_SOURCE_URL'));
       cy.matchImage({ matchAgainst: image.path });
     });
  });
});
```

Which of these APIs would fit better for your usecase @notdotscott? Personally I'm more for the 2nd option as it seems to be a bit more flexible in other use-cases - any valid screenshot path could be passed to the `matchAgainst`.

**Reply by notdotscott** [reply]
Hi @FRSgit, thanks very much for the reply! I agree, the second option definitely looks more flexible, and is more declarative than the first, so that one gets my vote 👍

**Reply by FRSgit** [reply]
@notdotscott - I've just published an alpha version containing this feature (Go released under [`@frsource/cypress-plugin-visual-regression-diff@3.0.0-alpha.1--match-against`](https://www.npmjs.com/package/@frsource/cypress-plugin-visual-regression-diff/v/3.0.0-alpha.1--match-against)). Can you give it a try?

If you have any feedback (that it works or not 😄) write it down in #146 please 🙏 
Example usage can be seen [here](https://github.com/FRSOURCE/cypress-plugin-visual-regression-diff/pull/146/files#diff-dfbaaddd07b5193f3c2a58a1ead0e123eabe4d2fdf99d4b7b3e290fb3d3c44d7R6-R9).

**Reply by FRSgit** [reply]
Hey, I've released the matchAgainst functionality in the new release - 3.0.0! 🎉
Please open a ticket if you find any issues 🙂
